idaes/core/solvers/petsc.py

```diff
@@ -666,6 +668,7 @@ def _unscale(self, m):
                 if s is not None:
                     for i, x in enumerate(self.vecs[vname]):
                         self.vecs[vname][i] = x/s
+                already_scaled.add(vname)
```

Comment:
Since `vname` is being used already, using it in this set isn't a huge deal, but I would rather we didn't use `str(var)` altogether. Name generation can be very time consuming for variables with large indexing sets. Using `id(var)` would be much faster.